File: src/backend/shoppingCartService/application/ApiResponseHandler.cs
using domain.responses;

namespace shoppingCartService.Models.Responses;

public static class ApiResponseHandler
{
    public static SuccessResponse SendSuccessResponse(int statusCode, object result)
    {
        var responseData = new SuccessResponse
        {
            IsSuccess = true,
            StatusCode = statusCode,
            Result = result
        };
        return responseData;
    }

    public static ErrorResponse SendErrorResponse(int statusCode, string error)
    {
        var responseData = new ErrorResponse
        {
            IsSuccess = false,
            StatusCode = statusCode,
            Error = error,
        };

        return responseData;
    }
}

File: src/backend/orderService/src/responses/apiResponseHandler.ts
import { OkResponse } from "./success";
import { ErrorResponse } from "./error";
import { Response } from "express";

export class ApiResponseHandler {
  static sendSuccessResponse<T>(
    res: Response,
    statusCode: number,
    result: T
  ): void {
    const response: OkResponse<T> = {
      isSuccess: true,
      statusCode,
      result,
    };
    res.status(statusCode).send(response);
  }

  static sendErrorResponse(
    res: Response,
    statusCode: number,
    error: string
  ): void {
    const response: ErrorResponse = {
      isSuccess: false,
      statusCode,
      error,
    };
    res.status(statusCode).send(response);
  }
}
